lib/rules/new-parens.js

```diff
@@ -65,14 +51,15 @@ module.exports = {
 
         return {
             NewExpression(node) {
-                let token = sourceCode.getTokenAfter(node.callee);
-
-                // Skip ')'
-                while (token && isClosingParen(token)) {
-                    token = sourceCode.getTokenAfter(token);
+                if (node.arguments.length !== 0) {
+                    return;  // shortcut: if there are arguments, there have to be parens
                 }
 
-                if (!(token && isOpeningParen(token) && isInRange(token, node))) {
+                const lastParen = sourceCode.getLastToken(node);
```

Comment:
I know this is the pickiest of nitpicks, but I think I'd prefer this variable to be `lastToken` (in case you actually don't get a closing parenthesis back). `hasLastParen` on the next line is the one that matters-- I wouldn't want maintainers to be confused about whether this is definitely a closing paren or not.